mne/gui/tests/test_stc.py

```diff
@@ -0,0 +1,148 @@
+# -*- coding: utf-8 -*-
+# Authors: Alex Rockhill <aprockhill@mailbox.org>
+#
+# License: BSD-3-clause
+
+import os.path as op
+import numpy as np
+from numpy.testing import assert_allclose
+
+import pytest
+
+import mne
+from mne.datasets import testing
+from mne.io.constants import FIFF
+
+data_path = testing.data_path(download=False)
+subject = 'sample'
+subjects_dir = op.join(data_path, 'subjects')
+fname_raw = op.join(data_path, 'MEG', 'sample', 'sample_audvis_trunc_raw.fif')
+fname_fwd_vol = op.join(data_path, 'MEG', 'sample',
+                        'sample_audvis_trunc-meg-vol-7-fwd.fif')
+fname_fwd = op.join(
+    data_path, 'MEG', 'sample', 'sample_audvis_trunc-meg-eeg-oct-4-fwd.fif')
+
+
+def _fake_stc(src_type='vol'):
+    """Fake a 5D source time estimate."""
+    rng = np.random.default_rng(11)
+    n_epochs = 3
+    info = mne.io.read_info(fname_raw)
+    info = mne.pick_info(info, mne.pick_types(info, meg='grad'))
+    if src_type == 'vol':
+        forward = mne.read_forward_solution(fname_fwd_vol)
+    else:
+        assert src_type == 'surf'
+        forward = mne.read_forward_solution(fname_fwd)
+    src = forward['src']
+    for this_src in src:
+        this_src['coord_frame'] = FIFF.FIFFV_COORD_MRI
+        this_src['subject_his_id'] = 'sample'
+    freqs = np.arange(8, 10)
+    times = np.arange(0.1, 0.11, 1 / info['sfreq'])
+    data = rng.integers(-1000, 1000, size=(n_epochs, len(info.ch_names),
+                                           freqs.size, times.size))
+    epochs_tfr = mne.time_frequency.EpochsTFR(
+        info, data, times=times, freqs=freqs)
+    stc_data = rng.integers(-1000, 1000, size=(n_epochs, forward['nsource'], 3,
+                                               freqs.size, times.size))
+    return stc_data, src, epochs_tfr
+
+
+@pytest.fixture
+def _stc_viewer(renderer_interactive_pyvistaqt):
+    from qtpy.QtWidgets import QApplication
+    from mne.gui._stc import SourceEstimateViewer
+    app = QApplication.instance()
+    if app is None:
+        app = QApplication(["Source Estimate Viewer"])
+    # Use a fixture to create these classes so we can ensure that they
+    # are closed at the end of the test
+    guis = list()
+
+    def fun(*args, **kwargs):
+        with pytest.warns(RuntimeWarning, match='`pial` surface not found'):
+            guis.append(SourceEstimateViewer(*args, **kwargs))
+        return guis[-1]
+
+    yield fun
+
+    for gui in guis:
+        try:
+            gui.close()
+        except Exception:
+            pass
```

Comment:
You should not need this `except Exception`, can you remove it? If not, it seems indicative of a problem. GUIs should always have a `.close` that works

What might help you avoid segfaults here is to have a `app.process_events()` called once or twice after each close. And ensure that `_ALL_PLOTTERS` from `pyvista` is empty when you do this -- it should check to make sure your `plotter` is getting cleaned up properly by the `.close()` event.

Reply:
This is copied from the ieeg location GUI testing script. I probably should look there at the close event and that might fix things.

Reply:
I hope that it was just a typo when I copy pasted the frequency slider to the time slider and forgot to change the name, we'll see maybe that will fix everything since `mne/gui/tests/test_core.py` and `mne/gui/tests/test_ieeg_locate.py` both work and they have the same close events. I think all the close events could do to be looked at, but maybe better all at once.

Reply:
Is this still on open question?

Reply:
What's with this? Is this resolved ?

Reply:
This is still an open question, I remember continuing discussion elsewhere but since it involves three different GUIs we should not add this complexity to this PR and instead follow-up in another.